produce_request.go

```diff
@@ -50,8 +67,41 @@ func (r *ProduceRequest) encode(pe packetEncoder) error {
 			if err != nil {
 				return err
 			}
+			if metricRegistry != nil {
+				for _, messageBlock := range msgSet.Messages {
+					// Is this a fake "message" wrapping real messages?
+					if messageBlock.Msg.Set != nil {
+						topicRecordCount += int64(len(messageBlock.Msg.Set.Messages))
+					} else {
+						// A single uncompressed message
+						topicRecordCount++
+					}
+					// Better be safe than sorry when computing the compression ratio
+					if len(messageBlock.Msg.Value) != 0 {
+						compressionRatio := float64(messageBlock.Msg.payloadSize) /
```

Comment:
Is this calculation correct? Compression ratios are usually given as `uncompressed / compressed` not the other way around. Or am I misunderstanding which value is which?

Perhaps you'd better rename `payloadSize` to `compressedPayloadSize` to be explicit about this.